testsuites/syncgateway/functional/tests/test_log_rotation_new.py

```diff
@@ -0,0 +1,609 @@
+
+    data = load_sync_gateway_config(sg_conf, cluster_hosts["couchbase_servers"][0], cluster_conf)
+
+    # set logKeys as string in config file
+    data['logging']["console"]["log_keys"] = "ABCD"
+    # create temp config file in the same folder as sg_conf
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp, indent=4)
+
+    # Stop sync_gateways
+    log_info(">>> Stopping sync_gateway")
+    sg_helper = SyncGateway()
+    sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+    try:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+    except ProvisioningError:
+        sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=sg_conf)
+        # Remove generated conf file
+        os.remove(temp_conf)
+        return
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+    pytest.fail("SG shouldn't be started!!!!")
+
+
+@pytest.mark.syncgateway
+@pytest.mark.logging
+@pytest.mark.parametrize("sg_conf_name", ["log_rotation_new"])
+def test_log_nondefault_logKeys_set(params_from_base_test_setup, sg_conf_name):
+    """Test to verify non default logKeys with any invalid area.
+    SG should work even with non existing logging area
+    (positive case)
+    """
+    cluster_conf = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    log_info("Using cluster_conf: {}".format(cluster_conf))
+    log_info("Using sg_conf: {}".format(sg_conf))
+
+    cluster_helper = ClusterKeywords()
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_conf)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_ip = host_for_url(sg_admin_url)
+
+    if get_sync_gateway_version(sg_ip)[0] < "2.1":
+        pytest.skip("Test NA for SG  > 2.1")
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    # read sample sg_conf
+    sg_one_url = cluster_hosts["sync_gateways"][0]["public"]
+    data = load_sync_gateway_config(sg_conf, cluster_hosts["couchbase_servers"][0], cluster_conf)
+
+    # "FAKE" not valid area in logging
+    data['logging']["console"]["logKeys"] = ["HTTP", "FAKE"]
+    # create temp config file in the same folder as sg_conf
+    temp_conf = "/".join(sg_conf.split('/')[:-2]) + '/temp_conf.json'
+
+    with open(temp_conf, 'w') as fp:
+        json.dump(data, fp, indent=4)
+
+    # Stop sync_gateways
+    log_info(">>> Stopping sync_gateway")
+    sg_helper = SyncGateway()
+    sg_helper.stop_sync_gateways(cluster_config=cluster_conf, url=sg_one_url)
+
+    # Start sync_gateways
+    sg_helper.start_sync_gateways(cluster_config=cluster_conf, url=sg_one_url, config=temp_conf)
+
+    # Remove generated conf file
+    os.remove(temp_conf)
+
```

Comment:
no assert for this test?

Reply:
Test will fail if SG does not start, which is the pass criteria.